src/MediaSensor/__tests__/index.test.tsx

```diff
@@ -0,0 +1,137 @@
+import {createElement as h} from 'react';
+import {mount, shallow} from 'enzyme';
+import {MediaSensor} from '..';
+
+const createMatchMedia = () => {
+  const matchMedia = jest.fn();
+  const matchMediaReturn = {
+    matches: [{}],
+    addListener: jest.fn(),
+    removeListener: jest.fn()
+  };
+
+  (window as any).matchMedia = matchMedia;
+  matchMedia.mockImplementation(() => matchMediaReturn);
+
+  return [matchMedia, matchMediaReturn];
+};
+
+describe('<MediaSensor>', () => {
+  it('exists', () => {
+    expect(MediaSensor).toBeInstanceOf(Function);
+  });
+
+  it('calls correct media query', () => {
+    const [matchMedia] = createMatchMedia();
+    let children: any = (jest.fn()) as any;
+    let wrapper = shallow(<MediaSensor query='(foobar)'>{children}</MediaSensor>);
+
+    expect(matchMedia).toHaveBeenCalledTimes(1);
+    expect(matchMedia).toHaveBeenCalledWith('(foobar)');
+  });
+
+  it('gets default media query match on mount', () => {
+    const [matchMedia] = createMatchMedia();
+    let children: any = (jest.fn()) as any;
+    let wrapper = shallow(<MediaSensor query='(foobar)'>{children}</MediaSensor>);
+
+    expect(matchMedia).toHaveBeenCalledTimes(1);
+    expect(matchMedia).toHaveBeenCalledWith('(foobar)');
+
+    expect(children).toHaveBeenCalledTimes(1);
+    expect(children).toHaveBeenCalledWith({matches: true});
+
+    (window as any).matchMedia = () => ({
+      matches: null,
+      addListener: jest.fn(),
+      removeListener: jest.fn()
+    });
+
+    children = jest.fn();
+    wrapper = shallow(<MediaSensor query='(foobar)' render={children} />);
+
+    expect(children).toHaveBeenCalledTimes(1);
+    expect(children).toHaveBeenCalledWith({matches: false});
+  });
+
+  it('re-renders on media query change', () => {
+    let listener;
+    const [matchMedia, matchMediaReturn] = createMatchMedia();
+
+    (matchMediaReturn as any).addListener.mockImplementation((list) => {
+      listener = list;
+    });
+
+    const MyComp = jest.fn();
+
+    MyComp.mockImplementation(() => null);
+
+    const wrapper = mount(<MediaSensor query='(foobar)' comp={MyComp}></MediaSensor>);
+
+    expect(MyComp).toHaveBeenCalledTimes(1);
+    expect(MyComp.mock.calls[0][0]).toEqual({matches: true});
```

Comment:
Did you consider using [`toHaveBeenCalledWith`](https://facebook.github.io/jest/docs/en/expect.html#tohavebeencalledwitharg1-arg2-)?

Reply:
Yes, but here I want to check only the first argument.